compositor_common/src/scene/builtin_transformations.rs

```diff
@@ -50,6 +57,38 @@ pub struct TextureLayout {
     pub rotation: Degree,
 }
 
+#[derive(Debug, SerializeDisplay, DeserializeFromStr, Clone, Copy, PartialEq, Eq, Default)]
+pub enum MirrorMode {
+    #[default]
+    Horizontal,
+    Vertical,
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error("\"mode\" field can only be set to \"vertical\" or \"horizontal\".")]
+pub struct MirrorModeParseError;
+
+impl Display for MirrorMode {
```

Comment:
I had to implement custom serialization and deserizliation becuase flatten and default don't work together